django_freeradius/base/admin_filters.py

```diff
@@ -0,0 +1,53 @@
+from django.contrib.admin import SimpleListFilter
+from django.db.models import Count
+from django.utils import timezone
+from django.utils.translation import ugettext_lazy as _
+
+
+class DuplicateListFilter(SimpleListFilter):
+    title = _('find duplicates')
+    parameter_name = 'duplicates'
+
+    def lookups(self, request, model_admin):
+        return (
+                ('username', _('username')),
+                ('value', _('value')),
+               )
+
+    def queryset(self, request, queryset):
+        value = self.value()
+        if not value:
+            return
+        u = queryset.values(value).annotate(Count('id'))\
+            .order_by().filter(id__count__gt=1)
+        if not u:
+            return
+        if self.value() == 'value':
+            for i in u:
+                q = queryset.model.objects.filter(value=i[self.value()])
+            return queryset.model.objects.filter(pk__in=[qi.pk for qi in q])
+        elif self.value() == 'username':
+            u = queryset.values(
+                self.value()).annotate(Count('id'))\
+                .order_by().filter(id__count__gt=1)
+            for i in u:
+                q = queryset.model.objects.filter(username=i[self.value()])
+            qs = queryset.model.objects.filter(pk__in=[qi.pk for qi in q])
+            return qs
+
+
+class ExpiredListFilter(SimpleListFilter):
+    title = _('find expired')
+    parameter_name = 'expired'
+
+    def lookups(self, request, model_admin):
+        return (
+                ('expired', _('expired')),
+                ('not_expired', _('not expired')),
+               )
+
+    def queryset(self, request, queryset):
```

Comment:
what about putting this logic in a model manager and calling the manager from this method?

That way we can reuse this logic in other views (eg: API) + it's easier to unit-test (models are easier to test)

Reply:
Yes I will, good hint... I'm developing this code like a thief! 
I'll "cover the coverage" test as well, excuse me if I'm asiding the unit test. First time for me in Code Coverage, great!

Reply:
I faced code coverage, I understood how to get the things working, I'll go ahead with more time, thank you for all.